From 2451f0d33dc5227d240e7b03059c463c6e115983 Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Mon, 16 May 2011 18:56:55 -0500
Subject: [PATCH 119/233] vlan: Reflect needed_headroom in vlan netdev

Extracted from vendor drop QorIQ-DPAA-SDK-V1-20110609-systembuilder.iso
0149-vlan-Reflect-needed_headroom-in-vlan-netdev.patch

The stack checks needed_headroom before sending packets on to the
net_device, but when the device is on a VLAN, the VLAN code kept the
default.  This meant that drivers have to always check if there's enough
room, rendering needed_headroom useless.  We can fix this by copying
needed_headroom into the vlan device.

Signed-off-by: Andy Fleming <afleming@freescale.com>
---
 net/8021q/vlan_dev.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/net/8021q/vlan_dev.c b/net/8021q/vlan_dev.c
index 29b6348..5f7888c 100644
--- a/net/8021q/vlan_dev.c
+++ b/net/8021q/vlan_dev.c
@@ -739,6 +739,7 @@ static int vlan_dev_init(struct net_device *dev)
 	} else {
 		dev->header_ops      = &vlan_header_ops;
 		dev->hard_header_len = real_dev->hard_header_len + VLAN_HLEN;
+		dev->needed_headroom = real_dev->needed_headroom;
 		if (real_dev->netdev_ops->ndo_select_queue)
 			dev->netdev_ops = &vlan_netdev_ops_sq;
 		else
-- 
1.7.0.4

